addons/mail/wizard/mail_compose_message.py

```diff
@@ -403,8 +404,11 @@ def onchange_template_id(self, template_id, composition_mode, model, res_id):
                     'type': 'binary',  # override default_type from context, possibly meant for another model!
                 }
                 attachment_ids.append(Attachment.create(data_attach).id)
-            if values.get('attachment_ids', []) or attachment_ids:
-                values['attachment_ids'] = [(5,)] + values.get('attachment_ids', []) + attachment_ids
+
+            values['attachment_ids'] = values.get('attachment_ids', []) + attachment_ids
+            if keep:
+                checksum = Attachment.browse(values['attachment_ids']).mapped('checksum')
+                values['attachment_ids'] = Attachment.search([('checksum','in', checksum)]).ids
```

Comment:
`checksum` doesn't have a unique constrain, so you'll get several IDs having the same checksum